wm/screens.go

```diff
@@ -72,105 +97,86 @@ func (xsp *x11ScreensProvider) ScreenForGeometry(x int, y int, width int, height
-		xproto.Setup(x.x.Conn()).DefaultScreen(x.x.Conn()).Root).Reply()
+	primary, err := randr.GetOutputPrimary(xsp.x.x.Conn(), root).Reply()
 	if err == nil {
-		primaryInfo, _ = randr.GetOutputInfo(x.x.Conn(), primary.Output, 0).Reply()
+		primaryInfo, _ = randr.GetOutputInfo(xsp.x.x.Conn(), primary.Output, 0).Reply()
 	}
 	primaryFound := false
-	var firstFoundMmWidth uint32
-	var firstFoundWidth uint16
-	i := 0
+	var tmpScreens []*desktop.Screen
 	for _, output := range resources.Outputs {
-		outputInfo, err := randr.GetOutputInfo(x.x.Conn(), output, 0).Reply()
+		outputInfo, err := randr.GetOutputInfo(xsp.x.x.Conn(), output, 0).Reply()
 		if err != nil {
 			fyne.LogError("Could not get randr output", err)
 			continue
 		}
 		if outputInfo.Crtc == 0 || outputInfo.Connection == randr.ConnectionDisconnected {
 			continue
 		}
-		crtcInfo, err := randr.GetCrtcInfo(x.x.Conn(), outputInfo.Crtc, 0).Reply()
+		crtcInfo, err := randr.GetCrtcInfo(xsp.x.x.Conn(), outputInfo.Crtc, 0).Reply()
 		if err != nil {
 			fyne.LogError("Could not get randr crtcs", err)
 			continue
 		}
-		if i == 0 {
-			firstFoundMmWidth = outputInfo.MmWidth
-			firstFoundWidth = crtcInfo.Width
+		insertIndex := -1
+		for i, screen := range tmpScreens {
+			if screen.X >= int(crtcInfo.X) && screen.Y >= int(crtcInfo.Y) {
+				insertIndex = i
+				break
+			}
+
+		}
+		if insertIndex == -1 {
+			tmpScreens = append(tmpScreens, &desktop.Screen{Name: string(outputInfo.Name),
+				X: int(crtcInfo.X), Y: int(crtcInfo.Y), Width: int(crtcInfo.Width), Height: int(crtcInfo.Height),
+				Scale: getScale(crtcInfo.Width, uint16(outputInfo.MmWidth))})
+			insertIndex = len(tmpScreens) - 1
+		} else {
+			tmpScreens = append(tmpScreens, nil)
+			copy(tmpScreens[insertIndex+1:], tmpScreens[insertIndex:])
+			tmpScreens[insertIndex] = &desktop.Screen{Name: string(outputInfo.Name),
+				X: int(crtcInfo.X), Y: int(crtcInfo.Y), Width: int(crtcInfo.Width), Height: int(crtcInfo.Height),
+				Scale: getScale(crtcInfo.Width, uint16(outputInfo.MmWidth))}
 		}
-		xsp.screens = append(xsp.screens, &desktop.Screen{Name: string(outputInfo.Name),
-			X: int(crtcInfo.X), Y: int(crtcInfo.Y), Width: int(crtcInfo.Width), Height: int(crtcInfo.Height)})
 		if primaryInfo != nil {
 			if string(primaryInfo.Name) == string(outputInfo.Name) {
 				primaryFound = true
-				xsp.primary = xsp.screens[i]
-				xsp.active = xsp.screens[i]
-				xsp.scale = getScale(crtcInfo.Width, outputInfo.MmWidth)
+				xsp.primary = tmpScreens[insertIndex]
+				xsp.active = tmpScreens[insertIndex]
 			}
 		}
-		i++
 	}
 	if !primaryFound {
-		xsp.primary = xsp.screens[0]
-		xsp.active = xsp.screens[0]
-		xsp.scale = getScale(firstFoundWidth, firstFoundMmWidth)
+		xsp.primary = tmpScreens[0]
+		xsp.active = tmpScreens[0]
 	}
+	xsp.screens = tmpScreens
 }
 
-func (xsp *x11ScreensProvider) setupSingleScreen(x *x11WM) {
+func (xsp *x11ScreensProvider) setupSingleScreen() {
+	xsp.single = true
+	xsp.screens = nil
```

Comment:
This line seems redundant, if the line below sets up a new slice instead of appending a screen to the nil slice